src/lib/kw_db.sh

```diff
@@ -203,13 +201,14 @@ function select_from()
   local table="$1"
   local columns="${2:-"*"}"
   local pre_cmd="$3"
-  local order_by="$4"
-  local flag=${5:-'SILENT'}
-  local db="${6:-$DB_NAME}"
-  local db_folder="${7:-$KW_DATA_DIR}"
+  local -n _condition_array="$4"
```

Comment:
About the comment of making `select_from` accept an empty `@_condition_array` value, this line will cause problems if the argument is empty, as Bash will complain:
```bash
local: `': not a valid identifier
```

I managed to alter the function to accept an empty value (without changing the behavior of when the value isn't empty) with the below:

```
diff --git a/src/lib/kw_db.sh b/src/lib/kw_db.sh
index 80169c4..3039752 100644
--- a/src/lib/kw_db.sh
+++ b/src/lib/kw_db.sh
@@ -201,7 +201,7 @@ function select_from()
   local table="$1"
   local columns="${2:-"*"}"
   local pre_cmd="$3"
-  local -n _condition_array="$4"
+  local _condition_array="$4"
   local order_by=${5:-''}
   local flag=${6:-'SILENT'}
   local db="${7:-"$DB_NAME"}"
@@ -224,8 +224,8 @@ function select_from()
 
   query="SELECT ${columns} FROM ${table} ;"
 
-  if [[ -n "${!_condition_array[*]}" ]]; then
-    where_clause="$(generate_where_clause '_condition_array')"
+  if [[ -n "$_condition_array" ]]; then
+    where_clause="$(generate_where_clause "$_condition_array")"
     query="${query::-2} WHERE ${where_clause} ;"
   fi
```

 As in `select_from` we don't do any associative array manipulations (only `generate_where_clause`, FWIU) we can only check to see if `_condition_array` is empty or not and treat it as a normal Bash variable (not a `local -n` reference), which won't complain about it being empty. However, you **must** pass an empty value, in case there is no `WHERE` clause (otherwise, this will cause an error).

I leave to you to decide if you want to use my approach, but confirm if this doesn't invalidates the non-empty cases.